README.md

```diff
@@ -35,10 +35,36 @@ following files:
   1. as final step, return the bare-metal machine to the CentOS CI for other
      users (it will be re-installed with a minimal CentOS environment again)
 
+- `e2e.groovy` is the Jenkins Pipeline responsible for running e2e tests on
+  a multi-node kubernetes cluster hosted on [Centos CI](https://wiki.centos.org/QaWiki/CI).
+  It verifies complete e2e functionalities for a corresponding Pull request
+  on [ceph-csi](https://github.com/ceph/ceph-csi/tree/master) repository.
+  It executes the following stages:
+
+  1. dynamically allocate a Jenkins Slave (`node('cico-workspace')`) with tools
+     and configuration to request a bare-metal machine.
+  2. checkout the `centos/ci` branch of the repository, which contains scripts
+     for provisioning and preparing the environment for running tests.
+  3. reserve a bare-metal machine with `cico` (configured on the Jenkins Slave);
+     retry if not immediately available.
+  4. provision the reserved bare-metal machine with additional tools and
+     dependencies to run the test (see `prepare.sh` below).
+  5. set up a multi-node k8s cluster on the bare-metal machine for performing
+     e2e tests (see `multi-node-k8s.sh` below).
+  6. deploy rook on the multi node kubernetes cluster.
+  7. run the e2e tests on the configured setup against the corresponding
+     pull request on the [ceph-csi](https://github.com/ceph/ceph-csi/tree/master)
```

Comment:
shorten the url here too, or even better: use labels and list the label+url at the bottom of the file

Reply:
Thanks, I'll make the changes